carapace-server/src/main/java/org/carapaceproxy/server/backends/BackendHealthManager.java

```diff
@@ -201,16 +204,30 @@ public boolean isAvailable(String id) {
     }
 
     public void reloadConfiguration(RuntimeServerConfiguration newConfiguration, EndpointMapper mapper) {
-        if (this.connectTimeout != newConfiguration.getConnectTimeout()) {
-            this.connectTimeout = newConfiguration.getConnectTimeout();
+        int newPeriod = newConfiguration.getHealthProbePeriod();
+
+        if (timer != null) {
+            timer.shutdown();
+        }
+
+        if (this.connectTimeout.get() != newConfiguration.getConnectTimeout()) {
+            this.connectTimeout.set(newConfiguration.getConnectTimeout());
             LOG.info("Applying new connect timeout " + this.connectTimeout + " ms");
         }
+
         this.mapper = mapper;
+
+        this.period.set(newPeriod);
+        LOG.info("Applying health probe period " + newPeriod + " ms");
+
+        if (timer != null) {
+            start();
```

Comment:
We can do better.
Just cancel the ScheduledFuture returned by scheduleAtfixedRate and schudule a new execution